### ISSUE TITLE
Removing flags

### ISSUE THREAD

**Issue by renra** [pboling/flag_shih_tzu#86]
Removing existing flags leads to an issue with the class methods. Here is a minimal example ...

Initial state ..

```ruby
class Spaceship < ActiveRecord::Base
  include FlagShihTzu

  has_flags 
            1 => :warpdrive,
            2 => :shields
end

s = Spaceship.create!
s.warpdrive = true
s.shields = true
s.save
s.flags # => 3

Spaceship.warpdrive.count # => 1, all is well
```

Now we change the flags to just ..

```ruby
class Spaceship < ActiveRecord::Base
  include FlagShihTzu

  has_flags 
            1 => :warpdrive
end

s = Spaceship.first
s.flags # => 3
s.warpdrive? # => true

# Here is the problem
Spaceship.warpdrive.count # => 0
```

**Comment by pboling** [comment]
Yes, that's most likely an unavoidable issue, unless someone wants to think really hard and figure out a fix.  The problem is these bit fields store a single value in binary that represents all values via a bit mask.  Removing flags isn't really possible.  If a flag becomes unused or deprecated you can rename it to mark is as such, but the data will still be present in the DB.

This is a tradeoff for using bitfields.

**Comment by pboling** [comment]
Actually I'll leave this open as a reminder to add more documentation about this.

**Comment by pboling** [comment]
Documentation of flag removal/deprecation added with regression specs in 7bcaf8fda70bfcb7cbbe363118267ac8b8e51180
